profiler/profile.go

```diff
@@ -159,8 +168,9 @@ var profileTypes = map[ProfileType]profileType{
 	},
 }
 
-func collectGenericProfile(name string, delta *pprofutils.Delta) func(p *profiler) ([]byte, error) {
+func collectGenericProfile(name string, pt ProfileType, delta *pprofutils.Delta) func(p *profiler) ([]byte, error) {
```

Comment:
So, I tried just making `collectGenericProfile` just take a `ProfileType` argument and drop the less type-safe `name` argument. However, this creates a initialization cycle between the `profileTypes` map and the `collectGenericProfile` function, since we still want to get the profile name for doing the lookup, and getting the name requires accessing `profileTypes`.

Not sure how much refactoring to do in this PR. I could give `ProfileType` a `name() string` method that just does a switch statement, and remove `Name` from `profileType`? Or leave it kind of awkward for now and refactor in a subsequent PR?

Reply:
Leave it awkward for now, it's ok. We'll come back to it.